### DISCUSSION TITLE
Time to eat notifications/reminders when prebolusing

### DISCUSSION THREAD

**Discussion by aimilushev** [NightscoutFoundation/xDrip#3790] [Ideas]
I'm a xDrip android app user and I am normally[ pre-bolusing](https://childrenwithdiabetes.com/caring-for-diabetes/what-is-a-pre-bolus/) my insulin. When eating starchy foods I normally put my insulin 10 minutes before eating in order not to get big blood sugar spikes. 10 minutes after preparing my meal and just looking at it is often a painful process. I open xdrip a few times to verify at what time I put my insulin and checking my watch regularly during this time. 
I was thinking that it will be nice if we have a built in notifications/reminders that prebolus interval has passed and it's time to eat.

How I feel this feature should look like:
 - For example in Settings / Alarms and alerts menu to add "Prebolus notifications" or something similar where you may enable/disable and potentially configure default prebolus wait time. 
 -  When adding insulin treatment in the app users may decide wether or not a notification should be shown after a specific period of time has passed. This period in miinutes may be configured at the time of adding the treatment (via widget similar to the time to eat dial) or a simplified version may be just to add enable/disable toggle for prebolus notifications based on a default prebolus interval from settings configuration
 - After the treatment is added and users have indicated that they want to receive a notification the app will schedule a timer and when the time has passed a notification would indicate that it's time to eat

I'm an android engineer and such implementation feels quite simple. So it's up to the app owners to decide if such enhancement makes sense and I can handle the coding part. Also any ideas of the UX are also welcome.

**Comment by Navid200** [comment]
Hi,

Thanks for the suggestion.
I am not the owner.  But, I spend a lot of time on xDrip as a user and a junior developer.  What I state here is only my opinion.  Nothing more.  

First, I wonder if you know that xDrip has reminders.   You can find it under the other options (ellipsis or 3-dot) menu button at the top right corner of the main screen.  

In my opinion, some parts of diabetes control are dependent on so many different factors that automating them could be impossible.
How fast insulin acts depends on how much exercise you did yesterday, what your mood is today, how much basal insulin you have left, how much food there is already in your digestive system, .....  

I do take insulin sometimes before I eat.  And I have experienced having a low after as well as having a high after.  So, even though I have chosen how long to wait myself and not relied on technology to tell me how long to wait, I have still made mistakes.

I would worry if I wanted to set a time to wait value ahead of time for all my bolus injections.  
In my opinion, this time will have to be decided for every instance by human and not by technology.

I have had diabetes for decades and I inject insulin to control it.  I have considered a closed loop system for years but cannot come to pull the trigger.  I cannot feel comfortable having enough insulin attached to me that could kill me and rely on a few lines of code not to inject it all at once!  

In my opinion xDrip should stay away from the domain of automated diabetes control because it could make us a target for litigation.  

I am aware of serious shortcomings in xDrip.  One example is that the missed reading alert ignores volume profile.  Imagine if you go to a very quiet place and you are embarrassed by an alert and you lower the phone volume to min.
Then, you go home and forget to raise the volume back up and go to bed.  xDrip stops having readings and issues a missed reading alert.  But, you never hear it because the missed reading alert does not use a volume profile.
You wake up in the morning and find out that you have had no reading for the past 7 hours.
I consider that a serious problem.

I intend to change the missed reading alert to respect the volume profile.
You say you are a programmer.  I wish we could put our limited resources (your time and expertise being a part of that) into addressing such serious shortcomings before taking leaps onto other domains.

I repeat that this is all just my opinion and in no way is an official xDrip response.
